### DISCUSSION TITLE
what determines the porosity change of the negative electrode?

### DISCUSSION THREAD

**Discussion by RobinBluehub** [pybamm-team/PyBaMM#2314] [Q&A] [answered]
Hi, what determines the porosity change of the negative electrode? Is there a formula? I didn't find the code in pybamm. The aging mechanism is SEI: "Outer SEI solvent diffusivity [m2.s-1]": 4.25E-21; Li_plating: "Dead lithium decay constant [s-1]": 7.75E-5, Lithium plating kinetic rate constant [m.s -1]": 1E-9; LAM: Negative/Positive electrode LAM constant proportional term [s-1]: 8.78e-8. The charging protocol adopts the standard charging protocol: 1C discharge, 0.5C charging. When I simulate the LG50 battery, When the battery capacity declines to 90.6%, the negative electrode porosity has decayed to 0, and the negative electrode porosity decays too fast.

**Comment by valentinsulzer** [comment, marked as answer]
It's in https://github.com/pybamm-team/PyBaMM/blob/develop/pybamm/models/submodels/porosity/reaction_driven_porosity.py

**Reply by RobinBluehub** [reply]
Thank you for your reply, how much is the Negative electrode roughness ratio usually taken? Will it be greater than 1?

**Reply by valentinsulzer** [reply]
If you don't turn on the sei in cracks option it will be 1

**Reply by RobinBluehub** [reply]
Hi,tinosulzer.I choose the  sei on  cracks option,L_sei_cr_n * (roughness_n - 1) will be negative?

**Reply by valentinsulzer** [reply]
The variable names in that link are the same as the variable names for plotting. You can plot individual variables and check for yourself.
